integration_tests/e2e/goal/archiveGoal.cy.ts

```diff
@@ -142,25 +195,40 @@ context('Archive a goal', () => {
     )
   })
 
-  it('should ask for confirmation and archive the goal then return to overview if choosing yes', () => {
+  it('Should be able to archive a goal successfully', () => {
     // Given
-    const prisonNumber = 'G6115VJ'
-    const goalReference = '10efc562-be8f-4675-9283-9ede0c19dade'
+    const { goalReference } = goalToArchive
+    cy.task('archiveGoal', { prisonNumber, goalReference })
     cy.signIn()
-    cy.visit(`/plan/${prisonNumber}/goals/${goalReference}/archive`)
-    const archiveGoalPage = Page.verifyOnPage(ArchiveGoalPage)
+    cy.visit(`/plan/${prisonNumber}/view/overview`)
+    const overviewPage = Page.verifyOnPage(OverviewPage)
+
+    // When
+    const archiveGoalPage = overviewPage //
+      .hasGoalsDisplayed()
+      .hasNumberOfGoals(2)
+      .clickArchiveButtonForFirstGoal()
+
     archiveGoalPage //
       .selectReason(ReasonToArchiveGoalValue.OTHER)
       .enterReason('Just because...')
       .submitPage()
 
-    // When
+    cy.task('getGoalsByStatus', {
```

Comment:
Do you think this adds any value to the cypress test? 🤔 
IE. you're re-setting up the `getGoalsByStatus` stub with fewer goals, so that the 2nd rendering of the Overview page has the "correct" goals, and then you can do the assertions on lines 229 and 230

I get it, but I wonder what it's adding to the cypress test 🤔 . In that, if the implementation were to change in some way and for whatever reason the call to archive the goal changed/failed, then according to the page rendering then it apparently would have worked.
I think the important assertion is the one that verifies the archive API was called with the expected data 👍 

What do you think? We dont do this "re-setting the stubs with updated data for the purpose of view" elsewhere - eg. we dont do it in the various tests that edit goals or inductions .... 🤔

Reply:
Yeah I tend to agree and wasn't really happy with this test. I kept it as it was what I was using to drive the changes in general and it's the only test that walks the whole flow of overview page > archive reason > confirm > overview page. Open to suggestions on how to improve it.